packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -507,17 +568,25 @@ function WorklistDataGrid(props) {
         onKeyDown={handleKeyDown}
         {...(activeIndex != null && { onMouseUp, onMouseMove, onMouseLeave: onMouseUp })}
       >
-        <ColumnHeader
-          columns={dataGridColumns}
-          headerHeight={columnHeaderHeight}
-          tableHeight={tableHeight}
-          tabStopColumnIndex={focusedRow.current === 0 ? focusedCol.current : undefined}
-          onColumnSelect={handleColumnSelect}
-          onResizeMouseDown={onResizeMouseDown}
-        />
-        <tbody>
-          {buildRows(rows)}
-        </tbody>
+        <ColumnContext.Provider
+          value={{
+            pinnedColumnsLength: hasSelectableRows ? pinnedColumns.length + 1 : pinnedColumns.length,
+            pinnedColumnOffsets,
+            pinnedColumnsTotalWidth,
```

Comment:
Should this all be managed by one state variable object?  It looks like the column offsets and total width are usually calculated together.  The pinned columns length should be able to be part of the same object as they all depend on the number of pinned columns.

Reply:
Removed extra state variables here:
c7e169957a8ae9422a6857961fade208b4bc8a31
ccf753f2a86fa8e72b05eb5a4a4f19199510e76e